src/lib/smart-transactions.ts

```diff
@@ -127,10 +133,30 @@ export const sendTransactionWithRetries = async (
     maxRetries: 0n,
   };
 
+  let timeout: number;
+  if (options.timeout) {
+    timeout = options.timeout;
+  } else {
+  switch (options.commitment) {
+    case "processed":
+      timeout = 5 * SECONDS;
+      break;
+    case "confirmed":
+      timeout = 15 * SECONDS;
+      break;
+    case "finalized":
+      timeout = 30 * SECONDS;
+      break;
+    default:
+      timeout = DEFAULT_TRANSACTION_TIMEOUT;
+      break;
```

Comment:
Good thinking! Make sure you update CHANGELOG.md with the changes (make an '*in progress* heading for the next release if you need to).